### ISSUE TITLE
manage-faircoin-account URL does not exist

### ISSUE THREAD

**Issue by bhaugen** [FreedomCoop/valuenetwork#50]
https://testocp.freedomcoop.eu/work/manage-faircoin-account/

**Comment by bhaugen** [comment]
It does exist, but requires the id of the account at the end, like this:
https://testocp.freedomcoop.eu/work/manage-faircoin-account/20/
But that has a bug. I don't think it is a bug in the OCP software. I've discussed this with @XaviP and I can't reproduce the same error that he sees internally. I think it is a problem with the testocp installation. So I am taking myself off as the assignee.

I could be wrong about this. Xavi, if you think I am, or that there's anything I can do to help, please let me know.

**Comment by XaviP** [comment]
I've been doing all kind of testing with the wallet and I don't find why ocp can't init() the wallet. In the shell_plus the wallet inits without problems, but django throws the error in this line:
```
https://github.com/FreedomCoop/valuenetwork/blob/master/faircoin_nrp/electrum_fair_nrp.py#L177
```
So it seems that django can't wait for the network to be up.

**Comment by bhaugen** [comment]
This is different from our telegram conversation yesterday, when you had a problem getting the wallet to init from shell_plus. How did you solve that problem? Or did it resolve itself some way?
```
>>> efn.init()
electrum directory /home/ocp/.electrum-fair
Error: Cannot read config file.
wallet path /home/ocp/.electrum-fair/wallets/default_wallet
[profiler] load_transactions 0.0001
```

>  django throws the error in this line.
So it seems that django can't wait for the network to be up.

I don't have that problem in the development environment. Maybe it takes longer or something else is getting in the way with Apache and Mod_wsgi?  

How did you zero in on that line of code? I don't see it in the error page. From logs or something?

Basically all that is happening now with django and the wallet is getting accurate balances for FairCoin addresses an accurate statuses for transactions. Everything else is going thru the cronjob. We would run those thru the cronjob, too, if need be. They really only need to be re-gotten after new transactions.

Got any other ideas?

**Comment by XaviP** [comment]
Yes, the wallet was complaining about the config file yesterday, but that is a generic config (located in /etc/electrum.conf) that doesn't block the access to the wallet, just a warning. So, if I init the wallet in shell_plus, the wallet is working (I can get the balance, etc...)
I don't know which is your enviroment, and why this is working good in yours and not in mine.

**Comment by bhaugen** [comment]
> I don't know which is your enviroment, and why this is working good in yours and not in mine.

I'm developing with the django builtin runserver, not apache. Sending transactions and creating new addresses did not work in my environment, either. That's why I changed them to the cronjob.

The error line you cited is in the midst of this code:
```
# wait until connected
    while (network.is_connecting() and (n < 100)):
        time.sleep(0.5)
        n = n + 1

    if not network.is_connected():
        logger.error("Can not init Electrum Network. Exiting.")
        sys.exit(1)
```

So it's not django raising that error, it's electrum_fair_nrp. The network did not get connected before the sleep timed out.

Could be either it takes longer in your environment, or maybe:
http://www.tutorialspoint.com/python/time_sleep.htm
> any caught signal will terminate the sleep() following execution of that signal's catching routine.

**Comment by bhaugen** [comment]
Creating new addresses and sending transactions in my environment were I think threading problems. This doesn't seem to be, but I am not sure.

**Comment by XaviP** [comment]
> The network did not get connected before the sleep timed out.

But the time that it's supposed to wait (about 50secs) is not the time rensponse in the website. The error is shown almost inmediatily, so it seems to me that it doesn't work time.wait() in the enviroment with apache.

**Comment by bhaugen** [comment]
> The error is shown almost inmediatily, so it seems to me that it doesn't work time.wait() in the enviroment with apache.

That's what I suspected from that web page about time.sleep. Can be interrupted by signals, in this case I suspect from Apache or mod-wsgi.

**Comment by XaviP** [comment]
Maybe, but I think that now the apache and wsgi configs are standard, nothing special on it. So I don't have any idea on how it has to be configured to solve this.
Also, I don't think it's a good practice that on installing ocp one needs a very specific apache or wsgi configuration.
I think everything can be easier if we change the way npr runs. As a daemon, we woudn't have any of this problems, and configuration on install would be simpler, as no cron routines needed.
We are always avoiding to run npr as a deamon, and I really don't understand why.

**Comment by bhaugen** [comment]
> We are always avoiding to run npr as a deamon, and I really don't understand why.

Try it. I don't understand how to do it and make it work with django.

**Comment by bhaugen** [comment]
I wasn't happy with the cronjob, I just couldnt get it to work with django any other way and wanted to get something done that could work.

**Comment by XaviP** [comment]
Me too. I'm going to send a SOS

**Comment by bhaugen** [comment]
Is this issue ready to close?

**Comment by XaviP** [comment]
Yes!
